pkg/cidata/cidata.TEMPLATE.d/boot/40-install-containerd.sh

```diff
@@ -8,14 +8,31 @@ fi
 # This script does not work unless systemd is available
 command -v systemctl >/dev/null 2>&1 || exit 0
 
-if [ ! -x /usr/local/bin/nerdctl ]; then
+# Extract bin/nerdctl and compare whether it is newer than the current /usr/local/bin/nerdctl (if already exists).
+# Takes 4-5 seconds. (FIXME: optimize)
```

Comment:
Idea for optimization:

```console
root@lima-default:/mnt/lima-cidata# gunzip -c nerdctl-full.tgz | head -c 512 | tar tv | awk '/ bin\/$/ {print $4}'
2022-03-03
root@lima-default:/mnt/lima-cidata# date -r /usr/local/bin/nerdctl "+%F"
2022-03-03
```

Then do simple string comparison on the dates.

This relies on a tar header block being 512 bytes (so `tar` will not throw an error), and `bin/` being the first entry in the tarball. I think these are reasonable assumptions, but am not totally sure. If the first line doesn't return a non-empty string, fall back on the expensive check by extracting the tarball.

Reply:
> I think these are reasonable assumptions, but am not totally sure. 

Not guaranteed. https://github.com/containerd/nerdctl/blob/20233c26d26f11ca73a9a775fba87ca884ab14d2/Makefile#L67

Reply:
Yeah, not guaranteed, but seems likely to be true, and with the fallback it should still work.

My only other idea would be to extract the nerdctl version from the download URL and then do a version comparison (pass it in as `LIMA_CIDATA_NERDCTL_VERSION`).

Still not guaranteed if somebody uses a custom download location that doesn't include the version, or worse, a different version, but you can argue that they will get what they deserve if they do this. 😄